service/policy/db/migrations/20240618000000_add_delete_cascade.sql

```diff
@@ -140,110 +140,122 @@ ON DELETE CASCADE;
 
 -- Do not cascade deletion of Attribute Definitions when their parent Namespace is deleted
 ALTER TABLE attribute_definitions
-DROP CONSTRAINT attribute_definitions_namespace_id_fkey_cascades;
-
+DROP CONSTRAINT IF EXISTS attribute_definitions_namespace_id_fkey_cascades;
+ALTER TABLE attribute_definitions
+DROP CONSTRAINT IF EXISTS attribute_definitions_namespace_id_fkey; -- Add this
 ALTER TABLE attribute_definitions
 ADD CONSTRAINT attribute_definitions_namespace_id_fkey
 FOREIGN KEY (namespace_id)
 REFERENCES attribute_namespaces (id);
 
 -- Do not cascade deletion of Attribute Values when their parent Definition is deleted
 ALTER TABLE attribute_values
-DROP CONSTRAINT attribute_values_attribute_definition_id_fkey_cascades;
-
+DROP CONSTRAINT IF EXISTS attribute_values_attribute_definition_id_fkey_cascades;
+ALTER TABLE attribute_values
+DROP CONSTRAINT IF EXISTS attribute_values_attribute_definition_id_fkey; -- Add this
 ALTER TABLE attribute_values
 ADD CONSTRAINT attribute_values_attribute_definition_id_fkey
 FOREIGN KEY (attribute_definition_id)
 REFERENCES attribute_definitions (id);
 
 -- Do not cascade deletion of Resource Mappings when their parent Attribute Value is deleted
 ALTER TABLE resource_mappings
-DROP CONSTRAINT resource_mappings_attribute_value_id_fkey_cascades;
-
+DROP CONSTRAINT IF EXISTS resource_mappings_attribute_value_id_fkey_cascades;
+ALTER TABLE resource_mappings
+DROP CONSTRAINT IF EXISTS resource_mappings_attribute_value_id_fkey; -- Add this
 ALTER TABLE resource_mappings
 ADD CONSTRAINT resource_mappings_attribute_value_id_fkey
 FOREIGN KEY (attribute_value_id)
 REFERENCES attribute_values (id);
 
 -- Do not cascade deletion of Subject Mappings when their parent Attribute Value is deleted
 ALTER TABLE subject_mappings
-DROP CONSTRAINT subject_mappings_attribute_value_id_fkey_cascades;
-
+DROP CONSTRAINT IF EXISTS subject_mappings_attribute_value_id_fkey_cascades;
+ALTER TABLE subject_mappings
+DROP CONSTRAINT IF EXISTS subject_mappings_attribute_value_id_fkey; -- Add this
 ALTER TABLE subject_mappings
 ADD CONSTRAINT subject_mappings_attribute_value_id_fkey
 FOREIGN KEY (attribute_value_id)
 REFERENCES attribute_values (id);
 
 -- Do not cascade deletion of FQNs when their parent objects are deleted
 ALTER TABLE attribute_fqns
-DROP CONSTRAINT attribute_fqns_namespace_id_fkey_cascades;
-
+DROP CONSTRAINT IF EXISTS attribute_fqns_namespace_id_fkey_cascades;
+ALTER TABLE attribute_fqns
```

Comment:
I could be confused here, but it seems like in this down migration we'd be adding this constraint rather than dropping it.